frontend/src/hooks/analytics/analytics.ts

```diff
@@ -0,0 +1,44 @@
+import Analytics from "analytics-node";
+import { uniqueId } from "lodash";
+import { getConfig } from "../../libraries/config";
+
+export interface Dict {
+  [key: string]: any;
+}
+
+let segment: Analytics;
+let anonymousId: string;
+
+function init() {
+  if (segment)
+    return console.log("Segment Analytics has already been initialized");
+  anonymousId = uniqueId(Date.now() + "_");
```

Comment:
Why do we need `uniqueId` for that?

Reply:
A value to use as an anonymous ID required by segment to track a user's actions

Reply:
No, I mean why not just `Date.now()`?
Why do we need this extra wrapper?
Why not `Date.now() + '_' + Math.random()`?

Reply:
`Math.random()` is a solid option